tests/unit/test_vtk.py

```diff
@@ -1,9112 +1,409 @@
+
+def test_constant_data(setup):
+    # Test of the Exporter functionality to distinguish between constant and non-constant
+    # data during exporting.
+
+    # Define grid
+    g = pp.StructuredTriangleGrid([3] * 2, [1] * 2)
+    g.compute_geometry()
+
+    # Define data
+    dummy_scalar = np.ones(g.num_cells) * g.dim
+    dummy_vector = np.ones((3, g.num_cells)) * g.dim
+
+    # Export data
+    save = pp.Exporter(
+        g,
+        setup.file_name,
+        setup.folder,
+    )
+    # Add additional constant data (cell centers)
+    save.add_constant_data([(g, "cc", g.cell_centers)])
+    save.write_vtu([("dummy_scalar", dummy_scalar), ("dummy_vector", dummy_vector)])
+
+    # Check that exported vtu files and reference files are the same
+    for appendix in ["2", "constant_2"]:
+        assert _compare_vtu_files(
+            f"{setup.folder}/{setup.file_name}_{appendix}.vtu",
+            f"{setup.folder_reference}/constant_data_test_grid_{appendix}.vtu",
+        )
+
+
+def test_fractures_2d(setup):
+    # Test of the export functionality of FractureNetwork2d.
+
+    # Define network
+    p = np.array([[0, 2, 1, 2, 1], [0, 0, 0, 1, 2]])
+    e = np.array([[0, 2, 3], [1, 3, 4]])
+    domain = {"xmin": -2, "xmax": 3, "ymin": -2, "ymax": 3}
+    network_2d = pp.FractureNetwork2d(p, e, domain)
+
+    # Define data
+    dummy_scalar = np.ones(network_2d.num_frac())
+    dummy_vector = np.ones((3, network_2d.num_frac()))
+    data = {"dummy_scalar": dummy_scalar, "dummy_vector": dummy_vector}
+
+    # Make directory if not existent
+    if not os.path.exists(setup.folder):
+        os.makedirs(setup.folder)
+
+    # Export data
+    network_2d.to_file(
+        setup.folder + setup.file_name + ".vtu",
+        data=data,
+    )
+
+    # Check that exported vtu file and reference file are the same
+    assert _compare_vtu_files(
+        f"{setup.folder}/{setup.file_name}.vtu",
+        f"{setup.folder_reference}/fractures_2d.vtu",
+    )
+
+
+def test_fractures_3d(setup):
+    # Test of the export functionality of FractureNetwork3d.
+
+    # Define network
+    f_1 = pp.PlaneFracture(np.array([[0, 1, 2, 0], [0, 0, 1, 1], [0, 0, 1, 1]]))
+    f_2 = pp.PlaneFracture(
+        np.array([[0.5, 0.5, 0.5, 0.5], [-1, 2, 2, -1], [-1, -1, 2, 2]])
+    )
+    domain = {"xmin": -2, "xmax": 3, "ymin": -2, "ymax": 3, "zmin": -3, "zmax": 3}
+    network_3d = pp.FractureNetwork3d([f_1, f_2], domain=domain)
+
+    # Define data
+    num_frac = len(network_3d._fractures)
+    dummy_scalar = [[1] for _ in range(num_frac)]
+    dummy_vector = [[np.ones(3)] for _ in range(num_frac)]
+    data = {"dummy_scalar": dummy_scalar, "dummy_vector": dummy_vector}
+
+    # Make directory if not existent
+    if not os.path.exists(setup.folder):
+        os.makedirs(setup.folder)
+
+    # Export data
+    network_3d.to_file(
+        setup.folder + setup.file_name + ".vtu",
+        data=data,
+    )
+
+    # Check that exported vtu file and reference file are the same
```

Comment:
Same comment as above regarding documentincg the gmsh dependency.

Reply:
See above.